rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/impl/WordCashableRepositoryImpl.kt

```diff
@@ -0,0 +1,118 @@
+package systems.ajax.englishstudytelegrambot.repository.impl
+
+import org.bson.types.ObjectId
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.data.redis.core.ReactiveRedisTemplate
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+import reactor.kotlin.core.publisher.toMono
+import systems.ajax.englishstudytelegrambot.entity.Word
+import systems.ajax.englishstudytelegrambot.repository.WordRepository
+
+@Repository
+@Suppress("TooManyFunctions" )
+class WordCashableRepositoryImpl(
+    val redisTemplate: ReactiveRedisTemplate<String, Word>,
+    @Qualifier("wordRepositoryImpl") val wordRepository: WordRepository
+) : WordRepository {
+
+    override fun saveNewWord(word: Word): Mono<Word> =
+        wordRepository.saveNewWord(word)
+            .flatMap {
+                redisTemplate.opsForValue().set(it.createKeyFindWordById(), word).thenReturn(it)
+            }
+
+    override fun updateWordTranslating(wordId: ObjectId, newWordTranslate: String): Mono<Word> =
+        wordRepository.updateWordTranslating(wordId, newWordTranslate)
+            .flatMap { redisTemplate.opsForValue().set(it.createKeyFindWordById(), it).thenReturn(it) }
+
+    override fun deleteWord(wordId: ObjectId): Mono<Word> =
+        wordRepository.deleteWord(wordId)
+            .flatMap { redisTemplate.opsForValue().delete(it.createKeyFindWordById()).thenReturn(it) }
+
+    override fun isWordBelongsToLibraryByWordId(wordId: ObjectId, libraryId: ObjectId): Mono<Boolean> =
+        getWord(wordId)
+            .map { it.libraryId == libraryId }
+            .flatMap {
+                if (it) {
+                    it.toMono()
+                } else {
+                    Mono.empty()
+                }
+            }
+            .switchIfEmpty {
```

Comment:
```suggestion
            .fitler { it }
            .switchIfEmpty {
```

it'll look better but I'm confused with overall logic in this method

Reply:
Done